python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -344,3 +351,47 @@ def _reuse_task_definition(self, task_definition, metadata, image, secrets):
 
     def _task_metadata(self):
         return default_ecs_task_metadata(self.ec2, self.ecs)
+
+    @property
+    def supports_check_run_worker_health(self):
+        return True
+
+    def check_run_worker_health(self, run: PipelineRun):
+
+        tags = self._get_run_tags(run.run_id)
+
+        if not (tags.arn and tags.cluster):
+            return CheckRunHealthResult(WorkerStatus.UNKNOWN, "")
+
+        tasks = self.ecs.describe_tasks(tasks=[tags.arn], cluster=tags.cluster).get("tasks")
+        if not tasks:
+            return CheckRunHealthResult(WorkerStatus.UNKNOWN, "")
+
+        t = tasks[0]
+
+        running_statuses = ["PROVISIONING", "PENDING", "ACTIVATING", "RUNNING"]
+        stopped_statuses = ["DEACTIVATING", "STOPPING", "DEPROVISIONING", "STOPPED"]
+
+        if t.get("lastStatus") in running_statuses:
+            return CheckRunHealthResult(WorkerStatus.RUNNING)
+        elif t.get("lastStatus") in stopped_statuses:
+
+            failed_containers = []
+            for c in t.get("containers"):
+                if c.get("exitCode") != 0:
+                    failed_containers.append(c)
+            if len(failed_containers) > 0:
+                if len(failed_containers) > 1:
+                    container_str = "Containers"
+                else:
+                    container_str = "Container"
+                return CheckRunHealthResult(
+                    WorkerStatus.FAILED,
+                    f"ECS task failed. Stop code: {t.get('stopCode')}. Stop reason {t.get('stopReason')}. "
+                    f"{container_str} {c.get('name') for c in failed_containers} failed."
+                    f"Check the logs for task {t.get('taskArn')} for details.",
+                )
+
```

Comment:
Maybe an additional `elif` for if ECS returns an unknown status? Like if their API changes without us noticing or if we get some malformed response or something?

Reply:
i think line 397 should take care of that but lmk if i'm misunderstanding your comment

Reply:
Maybe an extra unit test just to prove it's working since the conditional logic in the implementation gets a little tricky to reason through - set the last status to "boom" and assert it returns `WorkerStatus.UNKNOWN`.